test/export/test_export.py

```diff
@@ -3537,6 +3537,48 @@ def forward(self, arg1, arg2, a=1, b=2):
         kwargs3 = {"b": 1}
         self._test_export_same_as_eager(kw_func2, args, kwargs3)
 
+    def test_kwargs_reorder(self):
+        class M(torch.nn.Module):
+            def forward(self, *, x, y, z):
+                return x + y + z
+
+        ep = export(
+            M(), (), {"z": torch.ones(3), "y": torch.ones(3), "x": torch.ones(3)}
+        )
+        ep.module()(**{"z": torch.ones(3), "y": torch.ones(3), "x": torch.ones(3)})
+        ep.module()(z=torch.ones(3), y=torch.ones(3), x=torch.ones(3))
+        ep.module()(x=torch.ones(3), z=torch.ones(3), y=torch.ones(3))
+
+    def test_set_example_inputs(self):
+        class M(torch.nn.Module):
+            def forward(self, a, *, x, y, z):
+                return a, x + y + z
+
+        inp = (
+            (torch.ones(3),),
+            {"z": torch.ones(3), "y": torch.ones(3), "x": torch.ones(3)},
+        )
+        ep = export(M(), inp[0], inp[1])
+        ep.module()(*ep.example_inputs[0], **ep.example_inputs[1])
+
+        ep.example_inputs = (
+            (torch.ones(3),),
+            {"x": torch.ones(3), "z": torch.ones(3), "y": torch.ones(3)},
+        )
+        ep.module()(*ep.example_inputs[0], **ep.example_inputs[1])
+
+        with self.assertRaisesRegex(ValueError, "Example inputs should be a tuple"):
+            ep.example_inputs = (torch.ones(3),)
+
+        with self.assertRaisesRegex(ValueError, "Ran into a kwarg keyword mismatch"):
```

Comment:
does it work when the keyword has default values? e.g. `def forward(self, a, *, x, y, z=torch.ones(2))`. Now am I still required to provide a value for `z`?

Reply:
Lol just found out that dynamo does not support default tensor kwargs:
> torch._dynamo.exc.Unsupported: Tracing through optional input is not supported yet
But for non-strict, we will not treat the kwarg with a default value as an input, so you *cant* provide a value for z.

Reply:
> Lol just found out that dynamo does not support default tensor kwargs:
> 
> > torch._dynamo.exc.Unsupported: Tracing through optional input is not supported yet
> > But for non-strict, we will not treat the kwarg with a default value as an input, so you _cant_ provide a value for z.

lol I see.